dts: Use a hyphen as the delimiter for binding properties

rruuaanng opened the pull request:
Supplement the missing binding omitted during
the implementation of RFC-53506.

asmellby commented on doc/releases/migration-guide-4.2.rst:
```diff
@@ -102,6 +102,17 @@ GPIO
+PWM
+===
+
+* Renamed the devicetree property ``silabs,ch_prescaler`` to ``silabs,ch-prescaler``.
```
I don't believe these two PWM properties need migration guide entries, they were added to the tree after 4.1 release.

henrikbrixandersen reviewed (commented):
The migration guides should mention which bindings the changed properties occur in.

jhedberg reviewed (changes requested):
Changing my review to "Request changes" since there are a couple of unresolved review comments.

rruuaanng committed:
dts: Use a hyphen as the delimiter for binding properties

Supplement the missing binding omitted during
the implementation of RFC-53506.

Signed-off-by: James Roy <rruuaanng@outlook.com>

jhedberg commented:
@rruuaanng please fix the merge conflict with `migration-guide-4.2.rst`. Also, I don't think you should be changing the migration guide for the already released Zephyr 4.1 version.

henrikbrixandersen commented on doc/releases/migration-guide-4.2.rst:
```diff
@@ -196,6 +196,12 @@ Modem
+LED
+===
+
+* Renamed the devicetree property ``log_scale_en`` to ``log-scale-en``.
```
This needs to mention in which binding, the rename occurred. Also, why are the other renames not listed?

rruuaanng replied:
https://github.com/zephyrproject-rtos/zephyr/pull/87320#discussion_r2002968446

For other changes, they are in `migration-4.1.rst`.

henrikbrixandersen replied:
I don't understand. Why are they in the migration guide for v4.1 when the changes are going into v4.2?

rruuaanng replied:
I'm sorry, these were missed during my implementation :(

henrikbrixandersen commented:
@rruuaanng Is this still relevant?
